### PR TITLE
OKTA-363617: statemanager.revoke implementation

### PR DESCRIPTION
**Pull request by bryanapellanes-okta**
- Renamed OktaState to OktaStateManager
- Implemented OktaStateManager.revoke
- Cleaned up formatting in areas that were touched
- Added and updated tests where appropriate

### CODE REVIEW

File: Okta.Xamarin/Okta.Xamarin/OktaAppShell.xaml.cs [hunk 1]
```diff
@@ -13,11 +13,12 @@ public OktaAppShell()
             InitializeComponent();
             Routing.RegisterRoute(nameof(ItemDetailPage), typeof(ItemDetailPage));
             Routing.RegisterRoute(nameof(NewItemPage), typeof(NewItemPage));
+            Routing.RegisterRoute(nameof(DiagnosticsPage), typeof(DiagnosticsPage));
         }
 
         private async void OnSignOutClicked(object sender, EventArgs e)
         {
-			OktaContext.Current.SignOut();
+            _ = OktaContext.Current.SignOutAsync();
```
Comment (laura-rodriguez) [inline Okta.Xamarin/Okta.Xamarin/OktaAppShell.xaml.cs:21]: Should this method be awaited?

File: Okta.Xamarin/Okta.Xamarin/OktaAppShell.xaml.cs [hunk 2]
```diff
@@ -13,11 +13,12 @@ public OktaAppShell()
             InitializeComponent();
             Routing.RegisterRoute(nameof(ItemDetailPage), typeof(ItemDetailPage));
             Routing.RegisterRoute(nameof(NewItemPage), typeof(NewItemPage));
+            Routing.RegisterRoute(nameof(DiagnosticsPage), typeof(DiagnosticsPage));
         }
 
         private async void OnSignOutClicked(object sender, EventArgs e)
```
Comment (laura-rodriguez) [inline Okta.Xamarin/Okta.Xamarin/OktaAppShell.xaml.cs:19]: Nit: Use  Async suffix.

File: Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs [hunk 3]
```diff
@@ -4,19 +4,24 @@
...
-    public class OktaState
+    public class OktaStateManager
     {
+        private HttpClient client = new HttpClient();
+
         /// <summary>
         /// Gets the token type.
         /// </summary>
-        public string TokenType { get; private set; }
+        public string TokenType { get; private set; } // TODO: delete this
```
Comment (laura-rodriguez) [inline Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs:24]: Is this TODO still relevant?

File: Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs [hunk 4]
```diff
@@ -42,22 +47,24 @@ public class OktaState
...
+        /// Initializes a new instance of the <see cref="OktaStateManager"/> class.
         /// </summary>
         /// <param name="accessToken"></param>
         /// <param name="tokenType"></param>
         /// <param name="idToken"></param>
         /// <param name="refreshToken"></param>
         /// <param name="expiresIn"></param>
         /// <param name="scope"></param>
-        public OktaState(string accessToken, string tokenType, string idToken = null, string refreshToken = null, int? expiresIn = null, string scope = null)
+        public OktaStateManager(string accessToken, string tokenType, string idToken = null, string refreshToken = null, int? expiresIn = null, string scope = null)
```
Comment (laura-rodriguez) [inline Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs:67]: This constructor has too many params, I wonder if it'd make sense to create a convenience class with these attributes. What do you think?

File: Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs [hunk 5]
```diff
@@ -80,6 +87,30 @@ public bool IsAuthenticated
             }
         }
 
+        /// <summary>
+        /// Gets or sets the last response received from the API.
+        /// </summary>
+        public HttpResponseMessage LastApiResponse { get; set; }
```
Comment (laura-rodriguez) [inline Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs:93]: Curious, what's the purpose of this property?

File: Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs [hunk 6]
```diff
@@ -80,6 +87,30 @@ public bool IsAuthenticated
...
+
+        /// <summary>
+        /// Gets the token of the specified type.
+        /// </summary>
+        /// <param name="tokenType">The token type.</param>
+        /// <returns>string</returns>
+        public string GetToken(TokenType tokenType)
+        {
+            switch (tokenType)
```
Comment (laura-rodriguez) [inline Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs:102]: Should we also include Id token?

**Review by laura-rodriguez — APPROVED** [review]
Left a few questions, but overall LGTM 👍

File: Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs (hunk 3 above)
Comment (bryanapellanes-okta) [reply Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs:24]: Yes, it isn't used but if I deleted it I would have to review a lot of code that references it, mostly code that calls the constructor.  I'll handle that in a separate PR.

File: Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs (hunk 4 above)
Comment (bryanapellanes-okta) [reply Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs:67]: I tend to agree; most of the parameters are optional.  I may just remove the optional parameters and use object initialization instead.

File: Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs (hunk 5 above)
Comment (bryanapellanes-okta) [reply Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs:93]: This is mostly for debugging.  I'll add that to the comment.

File: Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs (hunk 6 above)
Comment (bryanapellanes-okta) [reply Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs:102]: Yes, I noticed that it was missing in later changes.  It will be in a later PR.

**Pull request merged**